lua/nvim-treesitter/refactor/navigation.lua

```diff
@@ -48,6 +44,26 @@ function M.list_definitions(bufnr)
   api.nvim_command('copen')
 end
 
+function M.goto_adjacent_usage(bufnr, delta)
+  local bufnr = bufnr or api.nvim_get_current_buf()
+  local node_at_point = ts_utils.get_node_at_cursor()
+
+  local def_node, scope = locals.find_definition(node_at_point, bufnr)
+
+  if not node_at_point then return end
+  local usages = locals.find_usages(def_node, scope, bufnr)
+
+  local index = utils.index_of(usages, node_at_point)
+  if not index then return end
+
+  local target_index = (index + delta + #usages - 1) % #usages + 1
+  ts_utils.goto_node(usages[target_index])
+  return usages[target_index]
+end
+
+function M.goto_next_usage(bufnr) return M.goto_adjacent_usage(bufnr, 1) or vim.cmd('normal! *') end
+function M.goto_previous_usage(bufnr) return M.goto_adjacent_usage(bufnr, -1) or vim.cmd('normal! #') end
```

Comment:
Not sure if falling back to the original `*` is something expected. Since you may believe you are jumping to the next usage when you aren't

Reply:
Yeah, I can remove the fall back